### ISSUE TITLE
manage crawls tab gives error on prod box

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/was-registrar#11]
(Note:  this GUI is not currently used by @nullhandle;  he uses scripts instead)

https://was-registrar.stanford.edu/crawls/index  gives:

![was-registrar error](https://cloud.githubusercontent.com/assets/96775/16287606/45b65de2-3899-11e6-92e7-077947bd286a.png)

note that this is only on  was-registrar;   was-registrar-stage and was-registrar-dev work.   A config issue?   Not spotting it immediately ...

**Comment by drh-stanford** [comment]
We're missing configuration information for the APOs on DEV. From `config/environments/production.rb`

```
  config.crawl_apos = 'druid:wr005wn5739,druid:yf700yh0557,druid:xt299pt7593'
  config.seed_apo = 'druid:xt299pt7593'
```

**Comment by drh-stanford** [comment]
```
Processing by CrawlsController#index as HTML
Completed 500 Internal Server Error in 108ms (ActiveRecord: 0.0ms)

NoMethodError (undefined method `first' for nil:NilClass):
  lib/was/utilities/dor_utilities.rb:16:in `block in get_collections_list'
  lib/was/utilities/dor_utilities.rb:11:in `each'
  lib/was/utilities/dor_utilities.rb:11:in `get_collections_list'
  app/controllers/crawls_controller.rb:15:in `index'
```

**Comment by drh-stanford** [comment]
I did some debugging and it looks like the `crawl_apos` configured in production are wrong.

```
2.1.0 :010 > apo = apos_list.first
 => "druid:wr005wn5739" 
2.1.0 :011 >           resource = RestClient::Resource.new(
2.1.0 :012 >                 Rails.configuration.apo_list_call, timeout: 60, open_timeout: 60
2.1.0 :013?>             )
 => #<RestClient::Resource:0x00000004645a88 @url="http://dorfetcher-prod.stanford.edu/apos/", @block=nil, @options={:timeout=>60, :open_timeout=>60}> 
2.1.0 :014 >           response = resource[apo].get
 => "{\"counts\":{\"total_count\":0}}" 
2.1.0 :015 > apo = apos_list[1]
 => "druid:yf700yh0557" 
2.1.0 :016 >           response = resource[apo].get
 => "{\"counts\":{\"total_count\":0}}" 
2.1.0 :017 > apo = apos_list[2]
 => "druid:xt299pt7593" 
2.1.0 :018 >           response = resource[apo].get
 => "{\"counts\":{\"total_count\":0}}" 
```

Basically, `dorfetcher` is failing to retrieve those APOs...

**Comment by drh-stanford** [comment]
prod crawl apos should be: wr005wn5739 and yf700yh0557  (those two, only)
prod seed apo should be: xt299pt7593 (only)

**Comment by drh-stanford** [comment]
I changed the configuration in prod, but still get the same error:

```
2.1.0 :002 >     apo_reader = Was::Utilities::DorUtilities.new(Rails.configuration.crawl_apos)
 => #<Was::Utilities::DorUtilities:0x0000000552c3d0 @apos_string="druid:wr005wn5739,druid:yf700yh0557"> 
2.1.0 :003 >     apo_collection_list = apo_reader.get_collections_list
NoMethodError: undefined method `first' for nil:NilClass
	from /opt/app/was/was-registrar/releases/20151013213205/lib/was/utilities/dor_utilities.rb:16:in `block in get_collections_list'
	from /opt/app/was/was-registrar/releases/20151013213205/lib/was/utilities/dor_utilities.rb:11:in `each'
	from /opt/app/was/was-registrar/releases/20151013213205/lib/was/utilities/dor_utilities.rb:11:in `get_collections_list'
	from (irb):3
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/commands/console.rb:90:in `start'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/commands/console.rb:9:in `start'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/commands/commands_tasks.rb:69:in `console'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/commands/commands_tasks.rb:40:in `run_command!'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/commands.rb:17:in `<top (required)>'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/app_rails_loader.rb:43:in `require'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/app_rails_loader.rb:43:in `block in exec_app_rails'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/app_rails_loader.rb:32:in `loop'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/app_rails_loader.rb:32:in `exec_app_rails'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/lib/rails/cli.rb:5:in `<top (required)>'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/bin/rails:9:in `require'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/railties-4.1.11/bin/rails:9:in `<top (required)>'
	from /opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/bin/rails:23:in `load'
```

**Comment by drh-stanford** [comment]
So I talked with @lmcglohon , and found that `dorfetcher` is deprecated and we need to import `dor-services` and do a `Dor::Item.find` to get the APO data.

**Comment by drh-stanford** [comment]
The `get_collections_list` is used by the `/crawls/index` page, and the `get_items_list` is used by the "Sync" link on that index page.

**Comment by drh-stanford** [comment]
For collections, the Argo3 index might support a query like this -- return all Collections whose APO is xt299pt7593:

```
objectType_ssim:"collection" AND is_governed_by_ssim:"info:fedora/druid:xt299pt7593"
```

For items, the Argo3 index might support a query like this -- return all Items whose APO is xt299pt7593:

```
objectType_ssim:"item" AND is_governed_by_ssim:"info:fedora/druid:xt299pt7593"
```

**Comment by ndushay** [comment]
closed in favor of #29 and #30
